gcloud-java-compute/src/test/java/com/google/gcloud/compute/DiskTypeTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+import java.math.BigInteger;
+
+public class DiskTypeTest {
+
+  private static final BigInteger ID = BigInteger.valueOf(42L);
+  private static final String CREATION_TIMESTAMP = "2016-01-20T04:39:00.210-08:00";
+  private static final String DESCRIPTION = "description";
+  private static final String VALID_DISK_SIZE = "10GB-10TB";
+  private static final Long DEFAULT_DISK_SIZE_GB = 10L;
+  private static final DiskTypeId DISK_TYPE_ID = DiskTypeId.of("project", "zone", "diskType");
+  private static final String DELETED = "2016-01-20T04:39:00.210-08:00";
+  private static final String DEPRECATED = "2016-01-20T04:37:00.210-08:00";
+  private static final String OBSOLETE = "2016-01-20T04:38:00.210-08:00";
+  private static final DeprecationStatus.Status STATUS = DeprecationStatus.Status.DELETED;
+  private static final DeprecationStatus<DiskTypeId> DEPRECATION_STATUS =
+      new DeprecationStatus<>(DELETED, DEPRECATED, OBSOLETE, DISK_TYPE_ID, STATUS);
+  private static final DiskType DISK_TYPE = DiskType.builder()
+      .id(ID)
+      .diskTypeId(DISK_TYPE_ID)
+      .creationTimestamp(CREATION_TIMESTAMP)
+      .description(DESCRIPTION)
+      .validDiskSize(VALID_DISK_SIZE)
+      .selfLink(DISK_TYPE_ID.toUrl())
+      .defaultDiskSizeGb(DEFAULT_DISK_SIZE_GB)
+      .deprecationStatus(DEPRECATION_STATUS)
+      .build();
+
+  @Test
+  public void testBuilder() {
+    assertEquals(ID, DISK_TYPE.id());
+    assertEquals(DISK_TYPE_ID, DISK_TYPE.diskTypeId());
+    assertEquals(CREATION_TIMESTAMP, DISK_TYPE.creationTimestamp());
+    assertEquals(DESCRIPTION, DISK_TYPE.description());
+    assertEquals(VALID_DISK_SIZE, DISK_TYPE.validDiskSize());
+    assertEquals(DISK_TYPE_ID.toUrl(), DISK_TYPE.selfLink());
+    assertEquals(DEFAULT_DISK_SIZE_GB, DISK_TYPE.defaultDiskSizeGb());
+    assertEquals(DEPRECATION_STATUS, DISK_TYPE.deprecationStatus());
+  }
+
+  @Test
+  public void testToPbAndFromPb() {
```

Comment:
Up to you, but we could test that a Disk Type (or any other model object) is properly handled by to/fromPb when all optional fields are left unspecified (it would mostly catch potential NPEs).